scene/gui/tree.cpp

```diff
@@ -543,6 +543,20 @@ bool TreeItem::is_collapsed() {
 	return collapsed;
 }
 
+void TreeItem::set_visible(bool p_visible) {
+	if (visible == p_visible || !tree) {
```

Comment:
`!tree` shouldn't be here, because it makes it impossible to set the property when not inside tree.

Better put it after `visible =`.

Reply:
Now I noticed you just copied code from `set_collapsed()`. Seems like that method is wrong too.